lib/watir-webdriver/element_collection.rb

```diff
@@ -89,16 +89,37 @@ def to_a
     def elements
       @parent.is_a?(IFrame) ? @parent.switch_to! : @parent.send(:assert_exists)
 
-      @elements ||= locator_class.new(
-        @parent.wd,
-        @selector,
-        element_class.attribute_list
-      ).locate_all
+      element_validator = element_validator_class.new
+      selector_builder = selector_builder_class.new(@parent, @selector, element_class.attribute_list)
+      locator = locator_class.new(@parent, @selector, selector_builder, element_validator)
+
+      @elements ||= locator.locate_all
     end
 
-    # overridable by subclasses
     def locator_class
-      ElementLocator
+      Kernel.const_get("#{Watir.locator_namespace}::#{element_class_name}::Locator")
+    rescue NameError
```

Comment:
To use defined, we should first cast string to constant using `const_get` which raises `NameError`.

Another approach would be something like this, but I'm not sure if it's better than existing one since we make more than one call:

```ruby
def locator_class
  if Kernel.const_defined?("#{Watir.locator_namespace}::#{element_class_name}::Locator")
    Kernel.const_get("#{Watir.locator_namespace}::#{element_class_name}::Locator")
  else
    Kernel.const_get("#{Watir.locator_namespace}::Element::Locator")
  end
end
```